test/e2e/csc_tests.go

```diff
@@ -0,0 +1,133 @@
+	// Define variables to be used in test
+	targetNamespace := "nonexisting-namespace"
+	packages := "etcd,federationv2,mongodb,amq-streams,couchbase,descheduler,prometheus,svcat,clusterlogging,dynatrace,metering"
+
+	// Get test namespace
+	namespace, err := ctx.GetNamespace()
+	if err != nil {
+		return fmt.Errorf("Could not get namespace: %v", err)
+	}
+
+	// Create the operatorsource to download the manifests
+	testOperatorSource := &operator.OperatorSource{
+		TypeMeta: metav1.TypeMeta{
+			Kind: operator.OperatorSourceKind,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-operators",
+			Namespace: namespace,
+		},
+		Spec: operator.OperatorSourceSpec{
+			Type:              "appregistry",
+			Endpoint:          "https://quay.io/cnr",
+			RegistryNamespace: "marketplace_e2e",
+		},
+	}
+	err = createRuntimeObject(f, ctx, testOperatorSource)
+	if err != nil {
+		return err
+	}
+
+	// Check that we created the catalogsourceconfig
+	resultCatalogSourceConfig := &operator.CatalogSourceConfig{}
+	err = WaitForResult(t, f, resultCatalogSourceConfig, namespace, testResourceName)
+	if err != nil {
+		return err
+	}
+
+	// Delete the existing catalogsourceconfig
+	err = deleteRuntimeObject(f, resultCatalogSourceConfig, testResourceName, namespace)
+	if err != nil {
+		return err
+	}
+
+	// Create a new catalogsourceconfig with a nonexistant targetNamespace
+	testCSC := &operator.CatalogSourceConfig{
+		TypeMeta: metav1.TypeMeta{
+			Kind: operator.OperatorSourceKind,
+		}, ObjectMeta: metav1.ObjectMeta{
+			Name:      testResourceName,
+			Namespace: namespace,
+		},
+		Spec: operator.CatalogSourceConfigSpec{
+			TargetNamespace: targetNamespace,
+			Packages:        packages,
+		}}
+	err = createRuntimeObject(f, ctx, testCSC)
+	if err != nil {
+		return err
+	}
+
+	// Check that we created the catalogsourceconfig with an invalid targetNamespace
+	err = WaitForResult(t, f, resultCatalogSourceConfig, namespace, testResourceName)
+	if err != nil {
+		return err
+	}
+
+	// Check if the catalogsourceconfig phase and message are the expected values
+	expectedPhase := "Configuring"
+	expectedMessage := fmt.Sprintf("namespaces \"%s\" not found", targetNamespace)
+	result := pollUntilTrue(5*time.Second, 1*time.Minute, func() bool {
+		// catalogsourceconfig should always exist so no wait
+		err = WaitForResult(t, f, resultCatalogSourceConfig, namespace, testResourceName)
+		if resultCatalogSourceConfig.Status.CurrentPhase.Name == expectedPhase &&
+			resultCatalogSourceConfig.Status.CurrentPhase.Message == expectedMessage {
+			return true
+		}
+		return false
+	})
+
+	if !result {
+		return fmt.Errorf("CatalogSourceConfig never reached expected phase/message, expected %v/%v got %v/%v", expectedPhase, expectedMessage, resultCatalogSourceConfig.Status.CurrentPhase.Name, resultCatalogSourceConfig.Status.CurrentPhase.Message)
+	}
+
+	// Create the nonexisting targetNamespace
+	err = createNamespace(f, ctx, targetNamespace)
```

Comment:
Does the `e2e` launched using the SDK use a super user account? Because the operator does not have `Roles` that allow it to interact with namespaces. This seems to imply that there is no way for us to verify if we are adding the correct `Roles` for the operator.

Reply:
@aravindhp There are two competing ideas here. the test client has similar super user rules that apply to it, similar to what `operator-sdk up` has (or akin to logging into oc as a super user), which is what is being used to create this namespace. That is separate from the operator itself having those roles, which are being tested here -- the operator is deployed using the deployment in the manifest/ folder when these tests are run in the e2e cluster.

Reply:
So your take is that the `ClusterOperator` will be marked as failing if the roles are incorrect? We should then update our deployment check to include that and not run any other tests if that fails.

Reply:
It should, but I think that's outside the scope of this test -- ClusterOperator having a failing status is a catch all, so we could lose granularity on why the error is happening if we just make that blanket assumption. Better for us to ignore ClusterOperator status, or at least just mark a failure and continue the tests. What if we want to run a test to purposely make ClusterOperator set a failed status? Or run a test to see a fail case that we would ignore because we are looking at ClusterOperator?

Reply:
Good point. I did not mean to have this be done as part of this PR. We should have a follow up PR for this. My desire is to have a test that will tell us if we messed up the `Roles` in the manifest folder for the operator.

Reply:
+1 I like the idea of explicit tests that check and ensure the state of OperatorStatus.